### PR TITLE
[BED-305] Return first match result when using ignoreMatches option

### PR DESCRIPTION
**Pull request by csucu**

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -262,10 +264,20 @@ func (p *parser) check() (Result, string, unused, error) {
 			p.fireMatch(token, result, s, ttl, err)
 			return result, s, unused{mechanisms[i+1:], redirect}, err
 		}
+
+		// Store the first match result if not already set
+		if p.ignoreMatches && matches && p.firstMatchResult == None {
+			p.firstMatchResult = result
+		}
```
Comment (dmotylev) [inline parser.go:271]: Why not also fireMatch()? as in L260:L264

File: parser_test.go [hunk 2]
```diff
@@ -228,9 +228,10 @@ func TestTokensSoritingHandleErrors(t *testing.T) {
 /* Test Parse.parse* methods here */
 
 type TokenTestCase struct {
-	Input  *token
-	Result Result
-	Match  bool
+	Input         *token
+	Result        Result
+	Match         bool
+	ignoreMatches bool
```
Comment (dmotylev) [inline parser_test.go:234]: Where do you use it?

Comment (csucu) [reply parser_test.go:234]: updated

File: parser.go (hunk 1 above)
Comment (csucu) [reply parser.go:271]: The purpose of fireMatch is to notify a listener when a mechanism matches, and in this case, we are deliberately choosing to ignore matches, so it wouldn't make sense to call to fireMatch() from my opinion. We are only storing the firstMatch for later reference .

**Review by illyasch — APPROVED** [review]

**Pull request merged**